operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/backup/ElasticsearchBackupRepository.java

```diff
@@ -303,29 +311,37 @@ private List<SnapshotInfo> findSnapshots(String repositoryName, Long backupId) {
     }
   }
 
-  private GetBackupStateResponseDto getBackupResponse(Long backupId, List<SnapshotInfo> snapshots) {
+  private GetBackupStateResponseDto getBackupResponse(
+      final Long backupId, final List<SnapshotInfo> snapshots) {
     final GetBackupStateResponseDto response = new GetBackupStateResponseDto(backupId);
+    final var firstSnapshot = snapshots.get(0);
     final Metadata metadata =
-        objectMapper.convertValue(snapshots.get(0).userMetadata(), Metadata.class);
+        objectMapper.convertValue(firstSnapshot.userMetadata(), Metadata.class);
     final Integer expectedSnapshotsCount = metadata.getPartCount();
+    final long startTimeInMilliseconds = firstSnapshot.startTime();
```

Comment:
We're always taking start time of first snapshot to count our timeout from. Should we may be in opposite take **end** time of **last** snapshot? Not to catch the cases when it took long to take individual snapshots.